src/framework/constraints.ml

```diff
@@ -12,12 +12,17 @@ module M = Messages
 
 (** Lifts a [Spec] so that the domain is [Hashcons]d *)
 module HashconsLifter (S:Spec)
-  : Spec with module D = Lattice.HConsed (S.D)
-          and module G = S.G
+  : Spec with module G = S.G
           and module C = S.C
```

Comment:
Are we not exposing D here anymore on purpose?

Reply:
Well yeah, it's clearly not needed (and probably the other constraints aren't either). I wouldn't be eager about removing all such constraints but there also isn't a reason to insist on keeping them since it causes duplication of these module definitions which can be quite long with multiple functor applications.

In fact, it cannot even be done here in the signature:
```
File "src/framework/constraints.ml", line 15, characters 48-54:
15 |   : Spec with module D = Lattice.HConsed (S.D) (struct let assume_idempotent = GobConfig.get_string "ana.int.refinement" = "never"
                                                     ^^^^^^
Error: Syntax error: module path expected.
```
Due to https://github.com/goblint/analyzer/pull/1186#discussion_r1363866695, the argument module cannot be defined anywhere before either without changing the behavior.